extensions/grpc/protoc/src/main/java/io/quarkus/grpc/protoc/plugin/MutinyGrpcGenerator.java

```diff
@@ -171,8 +174,31 @@ private MethodContext buildMethodContext(DescriptorProtos.MethodDescriptorProto
         return methodContext;
     }
 
-    private String lowerCaseFirst(String s) {
-        return Character.toLowerCase(s.charAt(0)) + s.substring(1);
+    static String adaptMethodName(String name) {
+        // We need to adjust the method name in the same way as the grpc-java compiler does:
+        // 1. Decapitalize
+        // 2. Replace an underscore and capitalize the following letter
+        // 3. Append the underscore for a java keyword
+        // (these rules are bit odd but we need to follow them unless we get rid of the grpc-java compiler)
+        // https://github.com/quarkusio/quarkus/issues/27170
+        name = Introspector.decapitalize(name);
```

Comment:
I'm not quite sure whether it's legal (from JPMS point of view) to use `java.beans.Introspector` and `javax.lang.model.SourceVersion` in this method...

Reply:
The thing is that this code should only be used during the build when the grpc stubs are generated.

Reply:
But can easily copy the code in our own utils if necessary...

Reply:
Done.